crates/core/src/api.rs

```diff
@@ -282,35 +297,31 @@ pub struct InputFile {
     pub syntax_tree: String,
 }
 
+/// This represents the details of a match in a file, but is not directly a match result.
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
 #[serde(rename_all = "camelCase")]
-pub struct Match {
+pub struct FileMatch {
     #[serde(default)]
     pub messages: Vec<Message>,
     #[serde(default)]
     pub variables: Vec<VariableMatch>,
     pub source_file: String,
     #[serde(default)]
     pub ranges: Vec<Range>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
+#[serde(rename_all = "camelCase")]
+pub struct Match {
+    #[serde(flatten)]
```

Comment:
/grit new Be careful with `serde(flatten)`, it changes wasm bindgen behavior and causes a Map to be generated instead of a struct. See https://github.com/RReverser/serde-wasm-bindgen/issues/49, for more details.